x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackPlugin.java

```diff
@@ -248,6 +253,16 @@ public Settings additionalSettings() {
         components.add(getLicenseService());
         components.add(getLicenseState());
 
+        if (DiscoveryNode.isDataNode(environment.settings())) {
```

Comment:
This makes me wonder if we should make the index a system index first ? This way, we could start the maintenance service in the system index plugin. I don't think it will cause issue to move the index into a system index since we have all the protection in place with the async search user ?

Reply:
Do you want me to convert AsyncSearch plugin into SystemIndexPlugin or should I create a separate "Async" plugin for that?

Reply:
I think we can have a dedicated plugin since the goal is to start the service for all users of the index ?

Reply:
:+1: